pkg/image/validation/os.go

```diff
@@ -217,3 +219,31 @@ func validateUnattended(def *image.Definition) []FailedValidation {
 
 	return failures
 }
+
+func validateRawConfig(def *image.Definition) []FailedValidation {
+	var failures []FailedValidation
+	isValidSize := regexp.MustCompile(`^([1-9]\d+|[1-9])+[MGT]`).MatchString
+
+	if def.Image.ImageType != image.TypeRAW && def.OperatingSystem.RawConfiguration.DiskSize != "" {
```

Comment:
Nit: I'd include an "exit early" block instead of repeating the check:

```
if def.OperatingSystem.RawConfiguration.DiskSize == "" {
    return
}
```